keadm/cmd/keadm/app/cmd/util/edgecoreinstaller.go

```diff
@@ -122,6 +123,19 @@ func (ku *KubeEdgeInstTool) createEdgeConfigFiles() error {
 	}
 	edgeCoreConfig.Modules.EdgeStream.TunnelServer = net.JoinHostPort(cloudCoreIP, strconv.Itoa(constants.DefaultTunnelPort))
 
+	if strings.TrimSpace(ku.Labels) != "" {
+		ku.Labels = strings.TrimSpace(ku.Labels)
+		labelsMap := make(map[string]string)
+		s1 := strings.Replace(ku.Labels, "{", "", -1)
+		s2 := strings.Replace(s1, "}", "", -1)
+		labels := strings.Split(s2, ",")
+		for _, label := range labels {
+			key := strings.Split(label, ":")[0]
+			value := strings.Split(label, ":")[1]
+			labelsMap[key] = value
+		}
+		edgeCoreConfig.Modules.Edged.Labels = labelsMap
+	}
```

Comment:
If the value of the label tag is like `{"key1":"value1", "key2":"value2"}`,then you can use the `json.Unmarshal`
```go
        if strings.TrimSpace(ku.Labels) != "" {
                labels := map[string]string{}
                if err := json.Unmarshal([]byte(ku.Labels), &labels); err != nil {
                        return fmt.Errorf("invalid Labels: %s", err)
                }
                edgeCoreConfig.Modules.Edged.Labels = labels
        }
```

But I recommend using `StringSliceVarP` to resolve the label flag

Reply:
thanks for your suggesstion